test-data/unit/check-super.test

```diff
@@ -115,3 +111,138 @@ class B(A):
     def foo(self):
         super(B, self).foo() # Not an error
 [out]
+
+[case testSuperWithAny]
+class B:
+    def f(self) -> None: pass
+class C(B):
+    def h(self, x) -> None:
+        reveal_type(super(x, x).f) # E: Revealed type is 'def ()'
+        reveal_type(super(C, x).f) # E: Revealed type is 'def ()'
+
+[case testSuperInUnannotatedMethod]
+class C:
+    def h(self):
+        super(C, self).xyz
+
+[case testSuperWithTypeType]
+from typing import Type
+
+class A:
+    def f(self) -> None: pass
+
+class B(A):
+    @classmethod
+    def g(cls, x) -> None:
+        reveal_type(super(cls, x).f) # E: Revealed type is 'def ()'
+[builtins fixtures/classmethod.pyi]
+
+[case testSuperWithTypeTypeAsSecondArgument]
+class B:
+    def f(self) -> None: pass
+
+class C(B):
+    def __new__(cls) -> 'C':
+        super(C, cls).f
+        return C()
+
+[case testSuperWithGenericSelf]
+from typing import TypeVar
+
+T = TypeVar('T', bound='C')
+
+class B:
+    def f(self) -> None: pass
+
+class C(B):
+    def f(self) -> None: pass
```

Comment:
I would make return types different here and in superclass (e.g. ``float`` and ``int``), so that we could be sure in ``reveal_type`` that the correct one is picked up.

Reply:
Updated.